Sources/HTTP/HTTPStreamingParser.swift

```diff
@@ -374,14 +374,16 @@ public class StreamingParser: HTTPResponseWriter {
             headers[.transferEncoding] = nil
         }
 
-        let availableConnections = maxRequests - (self.connectionCounter?.connectionCount ?? 0)
 
-        if clientRequestedKeepAlive && (availableConnections > 0) {
-            headers[.connection] = "Keep-Alive"
-            headers[.keepAlive] = "timeout=\(Int(StreamingParser.keepAliveTimeout)), max=\(availableConnections)"
-        } else {
-            headers[.connection] = "Close"
+        if clientRequestedKeepAlive {
+            let availableConnections = maxRequests - (self.connectionCounter?.connectionCount ?? 0)
+            if (availableConnections > 0) {
+                headers[.connection] = "Keep-Alive"
+                headers[.keepAlive] = "timeout=\(Int(StreamingParser.keepAliveTimeout)), max=\(availableConnections)"
+                return
+            }
         }
+        headers[.connection] = "Close"
```

Comment:
Put this in an `else` and remove the `return` line above.

Reply:
Unfortunately (I think) what your proposing breaks the original logic flow.

Reply:
I don't think it does... Which combination of `clientRequestedKeepAlive` and `availableConnections` do you think would give an incorrect result?

Reply:
Oh I see, if we request keepalive but there are no available connections.